### ISSUE TITLE
Mirror image for golang

### ISSUE THREAD

**Issue by alculquicondor** [kubernetes/k8s.io#5451]
The build/test jobs of the kueue subproject run in `eks-prow-build-cluster` and they are currently facing throttling from the docker registry due to the use of the `golang` base image.

We investigating using `kube-cross`, but it's too big for our needs https://github.com/kubernetes-sigs/kueue/issues/875

Is it possible to have a mirror of `golang` or a similar image in `registry.k8s.io`?

**Comment by alculquicondor** [comment]
Somehow it seems that the failures have reduced... Did something change in the cluster?

cc @trasc

**Comment by xmudrii** [comment]
> Did something change in the cluster?

No, everything is as it was at the time of creating this issue. It might be that Docker Hub images are not as requested as then, so we're not hitting the limit.

**Comment by dims** [comment]
@alculquicondor @xmudrii could we use the public ecr docker mirror?

`public.ecr.aws/docker/library/golang:1.18`

See https://gallery.ecr.aws/docker/library/golang for more info.

**Comment by dims** [comment]
```
$ docker pull public.ecr.aws/docker/library/golang:1.18
1.18: Pulling from docker/library/golang
c345c9e441f5: Pull complete
7b716680367d: Pull complete
0855378f8903: Pull complete
4bfb8dc93d41: Pull complete
530999854f46: Pull complete
6ad9f0b1b9e2: Pull complete
fc998f6bcdc2: Pull complete
Digest: sha256:50c889275d26f816b5314fc99f55425fa76b18fcaf16af255f5d57f09e1f48da
Status: Downloaded newer image for public.ecr.aws/docker/library/golang:1.18
public.ecr.aws/docker/library/golang:1.18
```
